boot.py

```diff
@@ -199,13 +199,18 @@ def on_new_window_async(self, w: sublime.Window) -> None:
     def on_pre_close_window(self, w: sublime.Window) -> None:
         windows.discard(w)
 
-    def on_post_move_async(self, view: sublime.View) -> None:
+    # Note: EventListener.on_post_move_async does not fire when a tab is moved out of the current window in such a way
+    # that a new window is created: https://github.com/sublimehq/sublime_text/issues/4630
+    # Hence, as a workaround we use on_pre_move, which still works in that case.
+    def on_pre_move(self, view: sublime.View) -> None:
         listeners = sublime_plugin.view_event_listeners.get(view.id())
         if not isinstance(listeners, list):
             return
         for listener in listeners:
             if isinstance(listener, DocumentSyncListener):
-                return listener.on_post_move_window_async()
+                # we need a small delay here, so that the DocumentSyncListener will recognize a possible new window
+                sublime.set_timeout_async(listener.on_post_move_window_async, timeout_ms=1)
```

Comment:
We usually just skip the timeout argument. Should be equivalent.

Reply:
Should I remove or leave it? I figured it would be nice to see at first glance what this parameter "1" stands for.

Reply:
You could skip the timeout value completely, including the number. I feel that that would be more meaningful since it would mean "as soon as possible after current message loop iteration finishes" instead of "after some arbitrary time".

In practice it would of course work the same pretty much.

Reply:
No, that is not possible, see my comment in the line above. If there is no delay, the `view.window()` will still point to the old window, which will result the logic in `DocumentSyncListener.on_post_move_window_async` not working.
ST needs a little time (>0, but <1ms) after on_pre_move was fired, to create the new window (or to attach the tab to an existing window). At least it didn't work when I tested it without the 1ms delay.

Reply:
With this 1ms delay we basically craft our own `on_post_move_async`, but without the bug from the same ST API function :)

Reply:
If those two behave differently then it doesn't bode well for the reliability of this solution. It means it relies on some arbitrary time period that might not be guaranteed in all cases.

Reply:
So what do you suggest then? Increase the delay to for example 5ms? Or keep the current behavior of LSP being mostly broken whenever a tab is dragged out of the window to create a new one?

Reply:
Tried this in safe mode on Mac and "no timeout" works the same as 1ms timeout in this simple case.
(Though I can imagine that different platforms might behave differently)

```
import sublime
import sublime_plugin

class Listener(sublime_plugin.EventListener):
    def on_pre_move(self, view):
    	print('on_pre_move before', view.window())
    	sublime.set_timeout_async(lambda: print('on_pre_move after timeout (0ms)', view.window()))
    	sublime.set_timeout_async(lambda: print('on_pre_move after timeout (1ms)', view.window()), 1)
```

on_pre_move before Window(3)
on_pre_move after timeout (0ms) Window(6)
on_pre_move after timeout (1ms) Window(6)

Reply:
I can confirm your results in safe mode on Windows, and surprisingly it seems to work for me in LSP without the delay as well now. But I can swear that when I tested this yesterday, that it didn't work for me without the delay. While I was debugging this, I even added a print statement in https://github.com/sublimelsp/LSP/blob/19a01aa045de04bcc805e56043923656548050e0/plugin/documents.py#L205-L206 and it would print everytime if, and only if I had not specified the delay.

When I use *Tools > Developer > Profile plugins*, it shows no other package under on_pre_move, which could possibly block the operation.

Not sure how we should proceed now. If we remove the delay, one option could be to keep the `is_enabled: return True` in the subclasses of LspTextCommand, for safety in case it doesn't work sometimes or for some users without the delay.